main.tf

```diff
@@ -195,14 +180,14 @@ resource "aws_security_group" "allow_443" {
     from_port   = 443
     to_port     = 443
     protocol    = "tcp"
-    cidr_blocks = var.secondary_cidr_blocks
+    cidr_blocks = length(var.secondary_cidr_blocks) > 0 ? var.secondary_cidr_blocks : [var.vpc_cidr]
   }
 
   egress {
     from_port   = 443
     to_port     = 443
     protocol    = "tcp"
-    cidr_blocks = var.secondary_cidr_blocks
+    cidr_blocks = length(var.secondary_cidr_blocks) > 0 ? var.secondary_cidr_blocks : [var.vpc_cidr]
```

Comment:
when there are no secondary CIDRs, this resource gets erroneously created, which keeps showing up as changes. This defaults to the original VPC CIDR range if no secondary CIDR exists